lib/ocpp/v201/profile.cpp

```diff
@@ -529,7 +529,62 @@ void update_itt(const int schedule_duration, std::vector<ChargingSchedulePeriod>
     }
 }
 
-CompositeSchedule calculate_composite_schedule(const CompositeSchedule& charging_station_max,
+/// \brief update the period based on the power limit
+/// \param[in] current the current startPeriod based on duration
+/// \param[inout] period the details of the current period in the schedule.
+/// \param[in] charging_schedule_period the period that is being compared to period.
+/// \param[in] current_charging_rate_unit_enum details of the current composite schedule charging_rate_unit for
+/// conversion.
+///
+ChargingSchedulePeriod adjust_charging_schedule_period(const int current, const ChargingSchedulePeriod period,
+                                                       const ChargingSchedulePeriod charging_schedule_period,
+                                                       const ChargingRateUnitEnum current_charging_rate_unit_enum) {
+    auto _period = period;
```

Comment:
For the sake of readability, could we name this something like `adjusted_period` to be more explicit?

Reply:
It's also unclear what the difference beween `period` and `charging_schedule_period` in the arguments we take is, given they're both `ChargingSchedulePeriod`s